crates/bevy_render/src/render_resource/resource_macros.rs

```diff
@@ -0,0 +1,108 @@
+// structs containing wgpu types take a long time to compile. this is particularly bad for generic
+// structs containing wgpu structs. we avoid that in debug builds (and for cargo check and rust analyzer)
+// by boxing and type-erasing with the `render_resource_wrapper` macro.
+// analysis from https://github.com/bevyengine/bevy/pull/5950#issuecomment-1243473071 indicates this is
+// due to `evaluate_obligations`. we should check if this can be removed after a fix lands for
+// https://github.com/rust-lang/rust/issues/99188 (and after other `evaluate_obligations`-related changes).
+#[cfg(debug_assertions)]
+#[macro_export]
+macro_rules! render_resource_wrapper {
+    ($wrapper_type:ident, $wgpu_type:ty) => {
+        #[derive(Clone, Debug)]
+        pub struct $wrapper_type(Option<std::sync::Arc<Box<()>>>);
+
+        impl $wrapper_type {
+            pub fn new(value: $wgpu_type) -> Self {
```

Comment:
If `$wgpu_type` is `!Send` or `!Sync`, won't this (and other `fn`'s that do `mem::transmute`) produce potentially unsound results, since `()` is `Send` and `Sync`?

Perhaps something like:

```suggestion
            pub fn new(value: $wgpu_type) -> Self where $wgpu_type: Send + Sync {
```

Could avoid that someone inadvertently uses this for something that isn't `Send + Sync` down the line?

Reply:
good catch, thanks!